### PR TITLE
Beaglebone debian dev

### PR DESCRIPTION
**Pull request by cdelston0**
I'm running gateway on a BeagleBone Enhanced board running a stock Debian image, and made these commits to support that.

I realise that the renaming of linux-raspbian -> linux-debian is potentially contentious and may need discussion on whether this is the preferred way to resolve this.

### CODE REVIEW

File: src/platforms/linux-debian.js [hunk 1]
```diff
@@ -158,7 +158,13 @@ function getWirelessMode() {
     mode = 'ap';
     enabled = true;
 
-    const data = fs.readFileSync('/etc/hostapd/hostapd.conf', 'utf8');
+    const data = null;
+    try {
+      data = fs.readFileSync('/etc/hostapd/hostapd.conf', 'utf8');
+    } catch (err) {
+      if (err.code !== 'ENOENT') throw err;
```
Comment (mrstegeman) [inline src/platforms/linux-debian.js]: Please use a full block:
```js
if (...) {
  ...
}
```

File: src/platforms/linux-debian.js [hunk 2]
```diff
@@ -158,7 +158,13 @@ function getWirelessMode() {
     mode = 'ap';
     enabled = true;
 
-    const data = fs.readFileSync('/etc/hostapd/hostapd.conf', 'utf8');
+    const data = null;
```
Comment (mrstegeman) [inline src/platforms/linux-debian.js]: This should be `let data`, not `const data`.

File: src/platform.js [hunk 3]
```diff
@@ -45,8 +45,10 @@ function getOS() {
   }
 
   const proc = child_process.spawnSync('lsb_release', ['-i', '-s']);
-  if (proc.status === 0 && proc.stdout.toString().trim() === 'Raspbian') {
-    return 'linux-raspbian';
+  if (proc.status === 0) {
+    var lsb_release = proc.stdout.toString().trim();
```
Comment (mrstegeman) [inline src/platform.js]: `const`, not `var`.

File: src/platform.js [hunk 4]
```diff
@@ -45,8 +45,10 @@ function getOS() {
...
+  if ((lsb_release === 'Debian') || (lsb_release === 'Raspbian')) {
```
Comment (mrstegeman) [inline src/platform.js]: Please indent this, and you're missing the closing brace.

**Comment by cdelston0** [comment]
Thankyou for your feedback. Javascript is not my mother tongue!

I believe I have implemented the requested changes, and I hope I haven't mangled the git history too much.

**Pull request merged**